src/node/test/historical_queries.cpp

```diff
@@ -48,50 +126,76 @@ void initialise_store(kv::Store& store, bool initialise_ledger_rekey = false)
     mi.status = ccf::MemberStatus::ACTIVE;
     mi.encryption_pub_key = crypto::make_rsa_key_pair()->public_key_pem();
     members->put(0, mi);
+    REQUIRE(tx.commit() == kv::CommitResult::SUCCESS);
   }
 
-  REQUIRE(tx.commit() == kv::CommitResult::SUCCESS);
+  return ts;
 }
 
-void initialise_ledger(
-  Ledger& ledger, std::shared_ptr<kv::StubConsensus> consensus)
+kv::Version write_transactions(kv::Store& kv_store, size_t tx_count)
 {
-  INFO("Rebuild ledger as seen by host");
-  auto next_ledger_entry = consensus->pop_oldest_entry();
-  while (next_ledger_entry.has_value())
+  const auto begin = kv_store.current_version();
+  const auto end = begin + tx_count;
+  for (size_t i = begin; i < end; ++i)
   {
-    const auto ib = ledger.insert(std::make_pair(
-      std::get<0>(next_ledger_entry.value()),
-      *std::get<1>(next_ledger_entry.value())));
-    REQUIRE(ib.second);
-    next_ledger_entry = consensus->pop_oldest_entry();
+    auto tx = kv_store.create_tx();
+    auto public_map = tx.rw<NumToString>("public:data");
+    auto private_map = tx.rw<NumToString>("data");
+    const auto s = std::to_string(i);
+    public_map->put(i, s);
+    private_map->put(i, s);
+
+    REQUIRE(tx.commit() == kv::CommitResult::SUCCESS);
   }
+
+  return kv_store.current_version();
 }
 
-void record_entry(kv::Store& store)
+kv::Version write_transactions_and_signature(
+  kv::Store& kv_store, size_t tx_count)
 {
-  auto idx = store.current_version() + 1;
+  write_transactions(kv_store, tx_count);
 
-  auto tx = store.create_tx();
-  auto public_map = tx.rw<NumToString>("public:data");
-  auto private_map = tx.rw<NumToString>("data");
-  const auto s = std::to_string(idx);
-  public_map->put(idx, s);
-  private_map->put(idx, s);
+  kv_store.get_history()->emit_signature();
+  kv_store.compact(kv_store.current_version());
+
+  return kv_store.current_version();
+}
 
-  LOG_DEBUG_FMT("Recording entry at {}", idx);
+kv::Version rekey(
+  kv::Store& kv_store,
+  const std::shared_ptr<ccf::LedgerSecrets>& ledger_secrets)
+{
+  // This isn't really used, but is needed for ShareManager, so can be recreated each time here
+  ccf::NetworkState network;
+  network.ledger_secrets = ledger_secrets;
+  ccf::ShareManager share_manager(network);
```

Comment:
(Totally out-of-scope for this PR) This pattern is a little odd. I think `ShareManager` should really be constructed with `ledger_secrets` directly, but it (like many other systems) relies on `network` to give it table definitions. We can replace every `tx.rw(network.foo)` with `tx.rw<Foo>(Tables::FOO)`, but its a bit riskier (needing to keep typename and table-name in sync, rather than using a pre-constructed pairing). I think we can get around this with statically named tables (so you could write `tx.rw<Nodes>()` for instance, since we know the name the `Nodes` table should _always_ have), but its not a priority fix right now.